pkg/storage/storage.go

```diff
@@ -198,19 +208,31 @@ func (p *storage) storePodMetrics(pods []PodMetricsPoint) {
 				klog.ErrorS(nil, "Got duplicate container point", "container", contPoint.Name, "pod", klog.KRef(podPoint.Namespace, podPoint.Name))
 				continue
 			}
+
+			var storedContPoint ContainerMetricsPoint
+			var foundStoredCont bool
+
+			storedPodPoint, foundPod := p.pods[podIdent]
+			if foundPod {
+				storedContPoint, foundStoredCont = storedPodPoint[contPoint.Name]
+			}
+
+			if foundStoredCont && contPoint.Timestamp.Before(storedContPoint.Timestamp) {
```

Comment:
Goal was to check contPoint vs prevPoint (not storedPoint). I would restore the orginal code add check in last check
Change:
```
					} else if prevPodPoint, found := p.prevPods[podIdent]; found {
						// Keep previous point
						prevContainers[contPoint.Name] = prevPodPoint[contPoint.Name]
					}
```
To
```
					} else if prevPodPoint, found := p.prevPods[podIdent]; found && prevPodPoint[contPoint.Name].Timestamp.Before(contPoint.Timestamp) {
						// Keep previous point
						prevContainers[contPoint.Name] = prevPodPoint[contPoint.Name]
					} else if found {
					  klog.InfoS(...)
					}

```